x-pack/platform/plugins/shared/streams/server/routes/streams/significant_events/read_significant_events.ts

```diff
@@ -38,30 +26,58 @@ export async function readSignificantEvents(
   const { assetClient, scopedClusterClient } = dependencies;
   const { name, from, to, bucketSize } = params;
 
-  const assetQueries = await assetClient.getAssetLinks(name, ['query']);
-  if (isEmpty(assetQueries)) {
+  const queryLinks = await assetClient.getAssetLinks(name, ['query']);
+  if (isEmpty(queryLinks)) {
     return [];
   }
 
-  const searchRequests = assetQueries.flatMap((asset) => {
-    return [
-      { index: name },
-      {
-        size: 0,
-        query: {
-          bool: {
-            filter: [
-              {
-                range: {
-                  '@timestamp': {
-                    gte: from.toISOString(),
-                    lte: to.toISOString(),
-                  },
-                },
+  const queryLinkByRuleId = keyBy(queryLinks, (queryLink) => getRuleIdFromQueryLink(queryLink));
+  const ruleIds = Object.keys(queryLinkByRuleId);
+
+  const response = await scopedClusterClient.asCurrentUser.search<
+    unknown,
+    {
+      by_rule: AggregationsTermsAggregateBase<{
+        key: string;
+        doc_count: number;
+        occurrences: AggregationsMultiBucketAggregateBase<{
+          key_as_string: string;
+          key: number;
+          doc_count: 0;
+        }>;
+        change_points: {
+          type: {
+            [key in ChangePointType]: { p_value: number; change_point: number };
+          };
+        };
+      }>;
+    }
+  >({
+    index: '.alerts-streams.alerts-default',
+    query: {
+      bool: {
+        filter: [
+          {
+            range: {
+              '@timestamp': {
+                gte: from.toISOString(),
+                lte: to.toISOString(),
               },
-              buildQuery(asset.query.kql.query),
-            ],
+            },
           },
+          {
+            terms: {
+              'kibana.alert.rule.uuid': ruleIds,
+            },
+          },
+        ],
+      },
+    },
+    aggs: {
+      by_rule: {
+        terms: {
+          field: 'kibana.alert.rule.uuid',
+          size: 10000,
```

Comment:
Do we want to put in a limit of queries per stream?

Reply:
you mean a limit to how many queries can be linked to a stream? maybe a question for @LucaWintergerst